File: app/src/main/java/com/example/testesantander/domain/contract/UserContract.java
package com.example.testesantander.domain.contract;

import com.example.testesantander.infra.BaseCallback;
import com.example.testesantander.login.LoginResponse;

public class UserContract {
    public interface IUserRepository{
        void login(String username, String password, BaseCallback<LoginResponse> onResult);
    }
}

File: BankApp/app/src/main/java/br/com/vinicius/bankapp/domain/user/UserContract.kt
package br.com.vinicius.bankapp.domain.user

import br.com.vinicius.bankapp.domain.user.User
import br.com.vinicius.bankapp.internal.BaseCallback

class UserContract {

    interface IUser{
        val username: String
        val password: String

        fun isValidEmpty(): Boolean

        fun validationPassword():Boolean

        fun validationCpf():Boolean

        fun validationEmail():Boolean

        fun startLogin(listener: BaseCallback<User>)
    }

    interface IRepository {
        fun startLogin(username: String, password: String, onResult: BaseCallback<User>)
    }
}